dl4j-examples/src/main/java/org/deeplearning4j/examples/nlp/sentencepiecernnexample/SentencePieceRNNExample.java

```diff
@@ -0,0 +1,260 @@
+
+
+        BertIterator b = BertIterator.builder()
+            .tokenizer(t)
+            .lengthHandling(BertIterator.LengthHandling.FIXED_LENGTH, 256)
+            .minibatchSize(32)
+            .sentenceProvider(new FileLabeledSentenceProvider(reviewFilesMap, rng))
+            .featureArrays(BertIterator.FeatureArrays.INDICES_MASK)
+            .vocabMap(t.getVocab())
+            .task(BertIterator.Task.SEQ_CLASSIFICATION)
+            .build();
+
+
+        return b;
+    }
+
+
+    public static void main(String[] args) throws Exception {
+
+
+        //Download and extract data
+        downloadData();
+
+
+        final int seed = 0;
+        //Seed for reproducibility
+        String pathToVocab = DATA_PATH + "vocab.txt";
+        // Path to vocab
+
+        // BertWordPieceTokenizerFactory initialized with given vocab
+        BertWordPieceTokenizerFactory t = new BertWordPieceTokenizerFactory(new File(pathToVocab), true, true, StandardCharsets.UTF_8);
+
+
+        MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
+            .seed(seed)
+            .updater(new Adam(1e-3))
+            .l2(1e-5)
+            .weightInit(WeightInit.XAVIER)
+            .list()
+            // matching EmbeddingSequenceLayer outputs with Bidirectional LSTM inputs
+            .setInputType(InputType.recurrent(1))
+//          // initialized weights with normal distribution, amount of inputs according to vocab size and off L2 for this layer
+            .layer(0, new EmbeddingSequenceLayer.Builder().weightInit(new NormalDistribution(0, 1)).l2(0)
+                .hasBias(true).nIn(t.getVocab().size()).nOut(128).build())
+//           // two Bidirectional LSTM layers in a row with dropout and tanh as activation function
+            .layer(new Bidirectional(new LSTM.Builder().nOut(256)
+                .dropOut(0.8).activation(Activation.TANH).build()))
+            .layer(new Bidirectional(new LSTM.Builder().nOut(256)
+                .dropOut(0.8).activation(Activation.TANH).build()))
+            .layer(new GlobalPoolingLayer(PoolingType.MAX))
+            // defining last layer with 2 outputs (2 classes - positive and negative),
+            // small dropout to avoid overfitting and MCXENT loss function
+            .layer(new OutputLayer.Builder().nOut(2)
+                .dropOut(0.97).activation(Activation.SOFTMAX)
+                .lossFunction(LossFunctions.LossFunction.MCXENT).build())
+            .build();
+
+
+        // Getting train and test BertIterators for both: test and train,
+        // changing argument isTraining: true to get train and false to get test respectively
+        BertIterator train = getBertDataSetIterator(true, t);
+        BertIterator test = getBertDataSetIterator(false, t);
+
+
+        // preprocessor for DataType matching
+        MultiDataSetPreProcessor mdsPreprocessor = new MultiDataSetPreProcessor() {
```

Comment:
Maybe note in the comment: "preprocessor for DataType matching - can be removed after 1.0.0-beta7 release"